picard/ui/mainwindow.py

```diff
@@ -1098,7 +1099,28 @@ def show_about(self):
         return AboutDialog.show_instance(self)
 
     def show_options(self, page=None):
-        return OptionsDialog.show_instance(page, self)
+        options_dialog = OptionsDialog.show_instance(page, self)
+        options_dialog.finished.connect(self.options_closed)
+        if self.script_editor_is_open:
+            # Disable signal processing to avoid saving changes not processed with "Make It So!"
+            self.script_editor_dialog.signal_save.disconnect()
+            self.script_editor_dialog.signal_selection_changed.disconnect()
+            self.script_editor_dialog.signal_update_scripts_list.disconnect()
+            self.script_editor_dialog.signal_index_changed.disconnect()
+
+            for op_page in options_dialog.pages:
```

Comment:
Instead of the search code here I would probably add a helper to `OptionsDialog`. Actually it can retrieve the page widget with a direct lookup, no need for a loop:

```python
    def get_page(self, name):
        return self.item_to_page[self.page_to_item[name]]
```

Then here it becomes:

```python
options_dialog.get_page('filerenaming').show_script_editing_page()
```